tools/cli/helpers/doc-parser/src/class-doc-parser.php

```diff
@@ -104,15 +146,304 @@ function ( $item ) use ( $ignore ) {
+			$hook_name = array_shift( $arguments );
+
+			// Purging any comments that could have been attributed to this argument.
+			$comments = $hook_name->getComments();
+			if ( ! empty( $comments ) ) {
+				$hook_name->setAttribute( 'comments', null );
+			}
+
+			$new_block = array(
+				'type'      => $node->name->name === 'apply_filters' ? 'filter' : 'action',
+				'line'      => $node->getLine(),
+				'end_line'  => $node->getEndLine() > 0 ? $node->getEndLine() : $node->getLine(),
+				'name'      => $this->pretty_print_hook_name( $hook_name ),
+				'arguments' => array(),
+			);
+
+			foreach ( $arguments as $argument ) {
+				$new_block['arguments'][] = $this->printer->prettyPrint( array( $argument ) );
+			}
+			$blocks[] = $new_block;
+		}
+
+		return $blocks;
+	}
+
+	/**
+	 * Pretty prints the name for the hook, taking an argument object as input.
+	 *
+	 * @suppress PhanRedundantConditionInLoop Phan is having trouble getting some of nikic/php-parser type definitions.
+	 *
+	 * @param Node\Arg $argument the first argument to the apply_filter or do_action call.
+	 * @return String pretty printed argument name.
+	 * @throws \Exception On an unexpected argument component.
+	 */
+	public function pretty_print_hook_name( Node\Arg $argument ): string {
+
+		if (
+			$argument->value instanceof Node\Scalar\String_
+				|| $argument->value instanceof Node\Expr\ConstFetch
+				|| $argument->value instanceof Node\Expr\ClassConstFetch
+		) {
+			return trim( $this->printer->prettyPrint( array( $argument ) ), '\'' );
+
+		} elseif ( $argument->value instanceof Node\Scalar\InterpolatedString ) {
+
+			$value = $argument->value;
+			'@phan-var Node\Scalar\InterpolatedString $value';
+
+			$result = '';
+
+			$parts = $value->parts;
+			'@phan-var array<string,Node\InterpolatedStringPart> $parts';
+
+			foreach ( $parts as $part ) {
+				if ( $part instanceof Node\InterpolatedStringPart ) {
+					$result .= $part->value;
+				} elseif ( $part instanceof Node\Expr ) {
+					$result .= '{' . $this->printer->prettyPrint( array( $part ) ) . '}';
+				} else {
+					throw new \Exception( 'Unexpected interpolated string component of type ' . get_class( $part ) );
+				}
+			}
+			return $result;
+
+		} elseif ( $argument->value instanceof Node\Expr\BinaryOp\Concat ) {
+
+			$value = $argument->value;
+			'@phan-var Node\Expr\BinaryOp\Concat $value';
+
+			$result = '';
+			foreach ( array( 'left', 'right' ) as $property ) {
+				$part = $value->{$property};
+				if ( $part instanceof Node\Scalar\String_ ) {
+					$result .= $part->value;
+				} elseif ( $part instanceof Node\Expr ) {
+					$result .= '{' . $this->printer->prettyPrint( array( $part ) ) . '}';
+				} else {
+					throw new \Exception( 'Unexpected concatenated string component of type ' . get_class( $part ) );
+				}
+			}
+			return $result;
 		}
 
-		return $output;
+		throw new \Exception( 'Unexpected function call argument of type ' . get_class( $argument->value ) );
```

Comment:
When we do the diagnostics, this should probably be turned into one instead of killing the whole process. 😉